nifi-nar-bundles/nifi-standard-bundle/nifi-standard-parameter-providers/src/main/java/org/apache/nifi/parameter/OnePasswordParameterProvider.java

```diff
@@ -0,0 +1,216 @@
+                                                                .header(CONTENT_TYPE, APPLICATION_JSON)
+                                                                .retrieve();
+
+                final JsonNode itemList = OBJECT_MAPPER.readTree(getVaultItems.body());
+                final Iterator<JsonNode> itemIterator = itemList.elements();
+
+                // we iterate though the items
+                while(itemIterator.hasNext()) {
+                    final JsonNode item = itemIterator.next();
+                    final String itemID = item.get("id").asText();
+
+                    final HttpResponseEntity getItemDetails = webClientServiceProvider.getWebClientService()
+                                                                    .get()
+                                                                    .uri(URI.create(connectServer + GET_VAULTS + "/" + vaultID + "/items/" + itemID))
+                                                                    .header(AUTHORIZATION_HEADER_NAME, AUTHORIZATION_HEADER_VALUE + accessToken)
+                                                                    .header(CONTENT_TYPE, APPLICATION_JSON)
+                                                                    .retrieve();
+
+                    final JsonNode itemDetails = OBJECT_MAPPER.readTree(getItemDetails.body());
+                    final String itemName = itemDetails.get("title").asText();
+                    final Iterator<JsonNode> itemFields = itemDetails.get("fields").elements();
+
+                    // we iterate through the fields
+                    while(itemFields.hasNext()) {
+                        final JsonNode field = itemFields.next();
+                        final String fieldId = field.get("id").asText();
+                        final JsonNode fieldValue = field.get("value");
+
+                        if(fieldValue != null) {
+                            final ParameterDescriptor parameterDescriptor = new ParameterDescriptor.Builder().name(itemName + "_" + fieldId).build();
+                            parameters.add(new Parameter(parameterDescriptor, fieldValue.asText(), null, true));
+                        }
+
+                    }
+
+                }
+
+                parameterGroups.add(new ParameterGroup(vaultName, parameters));
+            }
+
+        } catch (final IllegalArgumentException | IOException e) {
+            throw new RuntimeException("Failed to retrieve vault's items", e);
+        }
+
+        final AtomicInteger groupedParameterCount = new AtomicInteger(0);
+        final Collection<String> groupNames = new HashSet<>();
+        parameterGroups.forEach(group -> {
+            groupedParameterCount.addAndGet(group.getParameters().size());
+            groupNames.add(group.getGroupName());
+        });
+        getLogger().info("Fetched {} parameters.  Group names: {}", groupedParameterCount.get(), groupNames);
```

Comment:
Recommend minor adjustment for wording and spacing.
```suggestion
        getLogger().info("Fetched {} parameters with Group Names: {}", groupedParameterCount.get(), groupNames);
```